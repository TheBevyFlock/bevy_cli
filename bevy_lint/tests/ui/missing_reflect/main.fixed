// We want to require annotations for all errors, but don't need them for notes.
//@require-annotations-for-level: ERROR

#![feature(register_tool)]
#![register_tool(bevy)]
#![allow(dead_code)]
#![deny(bevy::missing_reflect)]

use bevy::prelude::*;

//~v NOTE: `Component` implemented here
#[derive(Component)]
#[derive(Reflect)]
struct MyComponent;
//~^ ERROR: defined a component without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived

//~v NOTE: `Resource` implemented here
#[derive(Resource)]
#[derive(Reflect)]
struct MyResource;
//~^ ERROR: defined a resource without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived

//~v NOTE: `Event` implemented here
#[derive(Event)]
#[derive(Reflect)]
struct MyEvent;
//~^ ERROR: defined an event without a `Reflect` implementation
//~| HELP: `Reflect` can be automatically derived
