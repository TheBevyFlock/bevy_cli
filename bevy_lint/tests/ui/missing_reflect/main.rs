#![feature(register_tool)]
#![register_tool(bevy)]
#![allow(dead_code)]
#![deny(bevy::missing_reflect)]

use bevy::{ecs::component::StorageType, prelude::*};

#[derive(Component)]
struct Missing;
//~^ ERROR: defined a component, resource, or event without a `Reflect` implementation

struct MissingImpl;
//~^ ERROR: defined a component, resource, or event without a `Reflect` implementation

impl Component for MissingImpl {
    const STORAGE_TYPE: StorageType = StorageType::Table;
}

#[derive(Component, Reflect)]
struct Satisfied;

#[allow(bevy::missing_reflect)]
#[derive(Component)]
struct Muted;

#[allow(bevy::missing_reflect)]
struct MutedImpl;

impl Component for MutedImpl {
    const STORAGE_TYPE: StorageType = StorageType::Table;
}
