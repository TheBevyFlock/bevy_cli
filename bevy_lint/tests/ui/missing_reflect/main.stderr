error: defined a component, resource, or event without a `Reflect` implementation
 --> tests/ui/missing_reflect/main.rs:9:8
  |
9 | struct Missing;
  |        ^^^^^^^
  |
note: the lint level is defined here
 --> tests/ui/missing_reflect/main.rs:4:9
  |
4 | #![deny(bevy::missing_reflect)]
  |         ^^^^^^^^^^^^^^^^^^^^^

error: defined a component, resource, or event without a `Reflect` implementation
  --> tests/ui/missing_reflect/main.rs:12:8
   |
12 | struct MissingImpl;
   |        ^^^^^^^^^^^

error: aborting due to 2 previous errors

